tool/smarts/src/main/java/org/openscience/cdk/isomorphism/Smarts.java

```diff
@@ -0,0 +1,2841 @@
+                            if (left.value() == BOND_UNSPEC || right.value() == BOND_UNSPEC)
+                                return new Expr(STEREOCHEMISTRY, 0);
+                            if (left.value() == right.value())
+                                return new Expr(STEREOCHEMISTRY, IStereoElement.TOGETHER);
+                            else
+                                return new Expr(STEREOCHEMISTRY, IStereoElement.OPPOSITE);
+                        default:
+                            return null;
+                    }
+                default:
+                    return null;
+            }
+            return null;
+        }
+
+        // final check
+        boolean finish() {
+            // check for unclosed rings, components, and branches
+            if (numRingOpens != 0 || curComponentId != 0 || !stack.isEmpty())
+                return false;
+            if (role != ReactionRole.None) {
+                if (role != ReactionRole.Agent)
+                    return false;
+                markReactionRoles();
+                for (IAtom atom : mol.atoms()) {
+                    ReactionRole role = atom.getProperty(CDKConstants.REACTION_ROLE);
+                    ((QueryAtom) atom).getExpression().and(
+                        new Expr(Expr.Type.REACTION_ROLE,
+                                 role.ordinal())
+                    );
+                }
+            }
+            // setup data structures for stereo chemistry
+            for (IAtom atom : astereo) {
+                if (local.get(atom) == null)
+                    continue;
+                IAtom[] ligands = new IAtom[4];
+                int     degree  = 0;
+                for (IBond bond : local.get(atom))
+                    ligands[degree++] = bond.getOther(atom);
+                // add implicit neighbor, and move to correct position
+                if (degree == 3) {
+                    ligands[degree++] = atom;
+                    for (int j = 3; j > 0; j--) {
+                        if (mol.indexOf(ligands[j]) > mol.indexOf(ligands[j - 1]))
+                            break;
+                        swap(ligands, j, j - 1);
+                    }
+                }
+                if (degree == 4) {
+                    // Note the left and right is stored in the atom expression, we
+                    // only need the IStereoElement for the local ordering of neighbors
+                    mol.addStereoElement(new TetrahedralChirality(atom, ligands, 0));
+                }
+            }
+            // convert SMARTS up/down bond stereo to something we use to match
+            if (!bstereo.isEmpty()) {
+                for (IBond bond : mol.bonds()) {
+                    Expr expr = ((QueryBond) BondRef.deref(bond)).getExpression();
+                    if (hasAliphaticDoubleBond(expr)) {
+                        IBond left = null, right = null;
+                        List<IBond> bBonds = local.get(bond.getB
```

Comment:
What does this method do? You use it in one of the tests below... does the smarts get read into an IQueryAtomContainer or so? Or is the smarts string added as content to the passed mol? (It will take me a bit more time to figure this out from the source code...)

Reply:
Which method? and aren't you two questions the same...?

> does the smarts get read into an IQueryAtomContainer

Yes

> Or is the smarts string added as content to the passed mol? 

Yes... but depends what you mean by added as content?

Reply:
See the other comment I just added: https://github.com/cdk/cdk/pull/429#discussion_r176315724